File: packages/ui/src/schemas/documents.ts

import { z } from 'zod';

export const DocumentResponseSchema = z.object({
    id: z.number(),
    filename: z.string(),
    status: z.string(),
    chunk_count: z.number(),
    page_count: z.number().nullable().optional(),
    file_size_bytes: z.number().nullable().optional(),
    error_message: z.string().nullable().optional(),
    created_at: z.string().nullable().optional(),
});

export const DocumentUploadResponseSchema = z.object({
    document_id: z.number(),
    filename: z.string(),
    status: z.string(),
    message: z.string(),
    embedding_error: z.string().nullable().optional(),
});

export const DocumentStatusResponseSchema = z.object({
    document_id: z.number(),
    filename: z.string(),
    status: z.string(),
    chunk_count: z.number(),
    page_count: z.number().nullable().optional(),
    error_message: z.string().nullable().optional(),
});

export type DocumentResponse = z.infer<typeof DocumentResponseSchema>;
export type DocumentUploadResponse = z.infer<typeof DocumentUploadResponseSchema>;
export type DocumentStatusResponse = z.infer<typeof DocumentStatusResponseSchema>;

File: packages/api/src/schemas/documents.py
"""Pydantic schemas for document management endpoints."""

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    """Response schema for a document."""

    id: int
    filename: str
    status: str
    chunk_count: int
    page_count: int | None = None
    file_size_bytes: int | None = None
    error_message: str | None = None
    created_at: str | None = None

    model_config = {"from_attributes": True}


class DocumentUploadResponse(BaseModel):
    """Response after uploading a document."""

    document_id: int
    filename: str
    status: str
    message: str
    embedding_error: str | None = None


class DocumentStatusResponse(BaseModel):
    """Response for document processing status."""

    document_id: int
    filename: str
    status: str
    chunk_count: int
    page_count: int | None = None
    error_message: str | None = None
